chan_handler.go

```diff
@@ -20,10 +22,11 @@ type StateChange struct {
 	To State
 }
 
-func NewChanHandler(scCh chan<- StateChange, errCh chan<- error) *ChanHandler {
+func NewChanHandler(logHandler LogPositionHandler, scCh chan<- StateChange, errCh chan<- error) *ChanHandler {
```

Comment:
Thinking about it, what would you think about a new creation API, or a way to set the LogPositionHandler?  A default handler could provide today's behavior, and we'd keep backward compatibility with existing users.